transitions/generate_patient_id_on_patients.js

```diff
@@ -0,0 +1,11 @@
+var transitionUtils = require('./utils');
+
+module.exports = {
+  filter: function(doc) {
+    return doc.type === 'person' &&
```

Comment:
> Doesn't this mean all person type docs will get a patient_id? I'm not sure if that's the intent because I don't have enough context but that's pretty odd behavior to me. Should there be a way to enable/disable which person docs have patient ids?

@mandric I tried to answer this in the Github comment linking to this pr:

> NB: this does mean we're going to generate ids for CHWs as well. I'm cool with this (as is Marc). I think honestly it could be useful if people want to be able to submit CHW reports in the future via SMS. The one down-side is that the field is patient_id. Ideally it would be person_id, but it's used in configs, and generally spread throughout the code, so if we're going to fix it I think it's going to be a separate piece of work.

> We also need to keep in mind that id generation is typically limited because they are short by design because they are used in SMS. So using them where if they are not used will be a waste. 

To expand on your concerns, the ratio of patients to CHWs is easily 10:1 or 100:1, so in terms of CHWs adding a strain on the ID system Marc and I didn't think it would be too much of a big deal. It also:
 - Greatly simplifies code and deployment. There is no universal rule where you can take a person in isolation and determine if they are a CHW or not: we would have to have configuration for SMS situations and App situations, and God-knows what when we have both.
 - Can be used in the future to support sending reports about CHWs via SMS. We already support creating reports about CHWs, and we have talked about utilising SMS in a number of ways, so I think this fits.

> Also other transitions will not complete (sync they all run synchronously in memory on a single change) until this one completes. So it potentially slows down the transition processing chain. 

Would this really be a concern? It's literally one write, which occurs if there is a new contact, and that contact was created via enketo.

> I'm in favor of having an easy way in configuration to enable/disable which documents are assigned patient ids, similar to how registrations behave by having a boolean expression that is evaluated, and by default it should be off. 

Right, so this would be the trying to keep it simple thing. Thinking about it though: I guess all we'd need is a couple of extra lines in `onChange` which takes a `bool_expr` configured and runs it against the doc to potentially filter it out? You can't do it in the `filter` function itself unfortunately, but I guess in `onChange` is fine?